arrow/extensions/geoarrow_point.go

```diff
@@ -0,0 +1,407 @@
+}
+
+// IsEmpty returns true if this is an empty point (all coordinates are NaN or zero with no dimension)
+func (p Point) IsEmpty() bool {
+	return math.IsNaN(p.X) && math.IsNaN(p.Y) && math.IsNaN(p.Z) && math.IsNaN(p.M)
+}
+
+// String returns a string representation of the point
+func (p Point) String() string {
+	if p.IsEmpty() {
+		return "POINT EMPTY"
+	}
+
+	switch p.Dimension {
+	case DimensionXY:
+		return fmt.Sprintf("POINT(%.6f %.6f)", p.X, p.Y)
+	case DimensionXYZ:
+		return fmt.Sprintf("POINT Z(%.6f %.6f %.6f)", p.X, p.Y, p.Z)
+	case DimensionXYM:
+		return fmt.Sprintf("POINT M(%.6f %.6f %.6f)", p.X, p.Y, p.M)
+	case DimensionXYZM:
+		return fmt.Sprintf("POINT ZM(%.6f %.6f %.6f %.6f)", p.X, p.Y, p.Z, p.M)
+	default:
+		return fmt.Sprintf("POINT(%.6f %.6f)", p.X, p.Y)
+	}
+}
+
+// PointType is the extension type for Point geometries
+type PointType struct {
+	arrow.ExtensionBase
+	metadata  *GeometryMetadata
+	dimension CoordinateDimension
+}
+
+// NewPointType creates a new Point extension type for 2D points
+func NewPointType() *PointType {
+	return NewPointTypeWithDimension(DimensionXY)
+}
+
+// NewPointTypeWithDimension creates a new Point extension type with specified dimension
+func NewPointTypeWithDimension(dim CoordinateDimension) *PointType {
+	metadata := NewGeometryMetadata()
+	coordType := createCoordinateType(dim)
+
+	return &PointType{
+		ExtensionBase: arrow.ExtensionBase{Storage: coordType},
+		metadata:      metadata,
+		dimension:     dim,
+	}
+}
+
+// NewPointTypeWithMetadata creates a new Point extension type with custom metadata
+func NewPointTypeWithMetadata(dim CoordinateDimension, metadata *GeometryMetadata) *PointType {
+	if metadata == nil {
+		metadata = NewGeometryMetadata()
+	}
+	coordType := createCoordinateType(dim)
+
+	return &PointType{
+		ExtensionBase: arrow.ExtensionBase{Storage: coordType},
+		metadata:      metadata,
+		dimension:     dim,
+	}
+}
+
+// ArrayType returns the array type for Point arrays
+func (*PointType) ArrayType() reflect.Type {
+	return reflect.TypeOf(PointArray{})
+}
+
+// ExtensionName returns the name of the extension
+func (*PointType) ExtensionName() string {
+	return "geoarrow.point"
+}
+
+// String returns a string representation of the type
+func (p *PointType) String() string {
+	return fmt.Sprintf("extension<%s[%s]>", p.ExtensionName(), p.dimension.String())
+}
+
+// ExtensionEquals checks if two extension types are equal
+func (p *PointType) ExtensionEquals(other arrow.ExtensionType) bool {
+	if p.ExtensionName() != other.ExtensionName() {
+		return false
+	}
+	if otherPoint, ok := other.(*PointType); ok {
+		return p.dimension == otherPoint.dimension
+	}
+	return arrow.TypeEqual(p.Storage, other.StorageType())
+}
+
+// Serialize serializes the extension type metadata
+func (p *PointType) Serialize() string {
+	if p.metadata == nil {
+		return ""
+	}
+	serialized, _ := p.metadata.Serialize()
+	return serialized
```

Comment:
personally I'd have the json.Marshal and such in here rather than having an explicit `Serialize` method on the metadata object.